From a0f215803f4739cdc2d7c596a5bed365220ac3ca Mon Sep 17 00:00:00 2001
From: Tom Deseyn <tom.deseyn@gmail.com>
Date: Mon, 16 Nov 2020 16:49:49 +0100
Subject: [PATCH] runtime: remove netcoreapp2.x targets

dagood: (6.0) changed patch to remove netcoreapp2.0 centrally in
src/libraries/Directory.Build.props instead of a find-replace across
all files.
---
 eng/empty.csproj                    |  1 +
 src/libraries/Directory.Build.props | 18 ++++++++++++++++++
 2 files changed, 19 insertions(+)

diff --git a/eng/empty.csproj b/eng/empty.csproj
index 6edde5507fc..68985aae3b6 100644
--- a/eng/empty.csproj
+++ b/eng/empty.csproj
@@ -10,6 +10,7 @@
 
   <PropertyGroup>
     <TargetFramework>netcoreapp2.0</TargetFramework>
+    <TargetFramework Condition="'$(DotNetBuildFromSource)' == 'true'">net5.0</TargetFramework>
   </PropertyGroup>
 
   <Import Project="versioning.targets" />
diff --git a/src/libraries/Directory.Build.props b/src/libraries/Directory.Build.props
index 3daf1a941be..94606e51460 100644
--- a/src/libraries/Directory.Build.props
+++ b/src/libraries/Directory.Build.props
@@ -45,6 +45,24 @@
     <IsShipping Condition="'$(IsTestProject)' == 'true' or '$(IsTestSupportProject)' == 'true'">false</IsShipping>
   </PropertyGroup>
 
+  <!-- If building from source, remove netcoreapp2.x from any project. -->
+  <PropertyGroup Condition="'$(DotNetBuildFromSource)' == 'true'">
+    <TargetFrameworks>;$(TargetFrameworks);</TargetFrameworks>
+
+    <TargetFrameworks>$(TargetFrameworks.Replace(';netcoreapp2.0;', ';'))</TargetFrameworks>
+    <TargetFrameworks>$(TargetFrameworks.Replace(';netcoreapp2.0-windows;', ';'))</TargetFrameworks>
+    <TargetFrameworks>$(TargetFrameworks.Replace(';netcoreapp2.0-Windows_NT;', ';'))</TargetFrameworks>
+    <TargetFrameworks>$(TargetFrameworks.Replace(';netcoreapp2.0-FreeBSD;', ';'))</TargetFrameworks>
+    <TargetFrameworks>$(TargetFrameworks.Replace(';netcoreapp2.0-Linux;', ';'))</TargetFrameworks>
+    <TargetFrameworks>$(TargetFrameworks.Replace(';netcoreapp2.0-OSX;', ';'))</TargetFrameworks>
+
+    <TargetFrameworks>$(TargetFrameworks.Replace(';netcoreapp2.1;', ';'))</TargetFrameworks>
+    <TargetFrameworks>$(TargetFrameworks.Replace(';netcoreapp2.1-Windows_NT;', ';'))</TargetFrameworks>
+    <TargetFrameworks>$(TargetFrameworks.Replace(';netcoreapp2.1-Unix;', ';'))</TargetFrameworks>
+
+    <TargetFrameworks>$(TargetFrameworks.Trim(';'))</TargetFrameworks>
+  </PropertyGroup>
+
   <!-- Warnings that should be disabled in our test projects. -->
   <PropertyGroup Condition="'$(IsTestProject)' == 'true' or '$(IsTestSupportProject)' == 'true'">
     <!-- don't warn on usage of BinaryFormatter from test projects -->
-- 
2.25.4

